mesonbuild/dependencies/base.py

```diff
@@ -404,10 +404,13 @@ def __init__(self, type_name: DependencyTypeName, environment: 'Environment', kw
         self.version_reqs: T.Optional[T.List[str]] = version_reqs
         self.required = kwargs.get('required', True)
         self.silent = kwargs.get('silent', False)
-        self.static = kwargs.get('static', self.env.coredata.get_option(OptionKey('prefer_static')))
-        self.libtype = LibType.STATIC if self.static else LibType.PREFER_SHARED
-        if not isinstance(self.static, bool):
+        static = kwargs.get('static')
+        if static is None:
+            static = self.env.coredata.get_option(OptionKey('prefer_static'))
+        if not isinstance(static, bool):
             raise DependencyException('Static keyword must be boolean')
```

Comment:
How is it possible for it to be something other than `(NoneType, bool)` now that we typecheck this?